lib/bolt/application.rb

```diff
@@ -0,0 +1,612 @@
+    # @param targets [Array[String]] The targets to run on.
+    # @param arguments [Array[String], NilClass] Arguments to pass to the script.
+    # @param env_vars [Hash] Environment variables to set on the target.
+    # @return [Bolt::ResultSet]
+    #
+    def script_run(script, targets, arguments: [], env_vars: {})
+      future = executor.future&.fetch('file_paths', false)
+      script = find_file(script, future)
+
+      Bolt::Util.validate_file('script', script)
+
+      with_benchmark do
+        executor.run_script(inventory.get_targets(targets), script, arguments, env_vars: env_vars)
+      end
+    end
+
+    # Generate a keypair using the configured secret plugin.
+    #
+    # @param force [Boolean] Forcibly create a keypair.
+    # @param plugin [String] The secret plugin to use.
+    # @return [Boolean]
+    #
+    def secret_createkeys(force: false, plugin: 'pkcs7')
+      unless plugins.by_name(plugin)
+        raise Bolt::Plugin::PluginError::Unknown, plugin
+      end
+
+      plugins.get_hook(plugin, :secret_createkeys)
+             .call('force' => force)
+    end
+
+    # Decrypt ciphertext using the configured secret plugin.
+    #
+    # @param ciphertext [String] The ciphertext to decrypt.
+    # @param plugin [String] The secret plugin to use.
+    # @return [Boolean]
+    #
+    def secret_decrypt(ciphertext, plugin: 'pkcs7')
+      unless plugins.by_name(plugin)
+        raise Bolt::Plugin::PluginError::Unknown, plugin
+      end
+
+      plugins.get_hook(plugin, :secret_decrypt)
+             .call('encrypted_value' => ciphertext)
+    end
+
+    # Encrypt plaintext using the configured secret plugin.
+    #
+    # @param plaintext [String] The plaintext to encrypt.
+    # @param plugin [String] The secret plugin to use.
+    # @return [Boolean]
+    #
+    def secret_encrypt(plaintext, plugin: 'pkcs7')
+      unless plugins.by_name(plugin)
+        raise Bolt::Plugin::PluginError::Unknown, plugin
+      end
+
+      plugins.get_hook(plugin, :secret_encrypt)
+             .call('plaintext_value' => plaintext)
+    end
+
+    # Run a task on a list of targets.
+    #
+    # @param task [String] The name of the task.
+    # @param options [Hash] Additional options.
+    # @return [Bolt::ResultSet]
+    #
+    def task_run(task, targets, params: {})
+      targets = inventory.get_targets(targets)
+
+      with_benchmark do
+        pal.run_task(task, targets, params, executor, inventory)
+      end
+    end
+
+    # TODO: This should probably be separate #show_task and #list_tasks methods.
+    #
+    # Show task information.
+    #
+    # @param task [String, NilClass] The name of the task to show.
+    # @param filter [String, NilClass] A string to filter tasks by.
+    # @return [Boolean]
+    #
+    def task_show(task, filter: nil)
+      if task
+        pal.get_task(task)
+      else
+        tasks = filter_content(pal.list_tasks(filter_content: true), filter)
```

Comment:
It'd be nice if this was `list_tasks_with_cache` just so that the task cache stays up to date...but strictly speaking it doesn't need to use the cache.